instrumentation/rspec/lib/opentelemetry/instrumentation/rspec/formatter.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry-api'
+
+module OpenTelemetry
+  module Instrumentation
+    module RSpec
+      # An RSpec Formatter that outputs Otel spans
+      class Formatter
+        attr_reader :output
+
+        ::RSpec::Core::Formatters.register self, :example_started, :example_finished, :example_group_started, :example_group_finished, :start, :stop
+
+        @clock = Time.method(:now)
+
+        def self.current_timestamp
+          @clock.call
+        end
+
+        def initialize(output = StringIO.new, tracer_provider = OpenTelemetry.tracer_provider)
+          @spans_and_tokens = []
+          @output = output
+          @tracer_provider = tracer_provider
+        end
+
+        def tracer
+          @tracer ||= @tracer_provider.tracer('RSpec')
```

Comment:
This would keep it consistent with the other auto instrumentation packages.
```suggestion
          @tracer ||= @tracer_provider.tracer('OpenTelemetry::Instrumentation::RSpec', OpenTelemetry::Instrumentation::RSpec::Version)
```

Reply:
sorted